frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -178,7 +188,25 @@ pub mod pallet {
         T::SmartContract,
         Twox64Concat,
         EraIndex,
-        EraStakingPoints<T::AccountId, BalanceOf<T>>,
+        EraStakingPoints<BalanceOf<T>>,
+    >;
+
+    /// Era in which dapp reward was last claimed.
+    #[pallet::storage]
+    #[pallet::getter(fn last_claimed_era)]
+    pub type LastClaimedEra<T: Config> =
```

Comment:
So, this is for dapps to do sequential claiming, right?

Reply:
Yes, I created this PR just to showcase how we could modify `claim_dapp` to avoid specifying era.
It shows lots of changes now because base branch changed...

I didn't want to push these changes into the original commit:
https://github.com/AstarNetwork/astar-frame/pull/6

Sorry for the confusion! I see that you've left comments here but they can be applied to the original commit, no problem.